be/src/util/arrow/block_convertor.cpp

```diff
@@ -408,7 +410,10 @@ Status FromBlockConverter::convert(std::shared_ptr<arrow::RecordBatch>* out) {
 Status convert_to_arrow_batch(const vectorized::Block& block,
                               const std::shared_ptr<arrow::Schema>& schema, arrow::MemoryPool* pool,
                               std::shared_ptr<arrow::RecordBatch>* result) {
-    FromBlockConverter converter(block, schema, pool);
+    // FromBlockConverter converter(block, schema, pool);
+    std::shared_ptr<arrow::Schema> block_arrow_schema;
+    RETURN_IF_ERROR(get_block_arrow_schema(block, &block_arrow_schema));
```

Comment:
Why not use schema from argument?

Reply:
after expr, schema may be changed. For example after a cast expr